### ISSUE TITLE
[BUG] history api returns empty result for certain item ID.

### ISSUE THREAD

**Issue by ha-ku** [Universalis-FFXIV/Universalis#442]
Querying [Market History API](https://universalis.app/api/history/LuXingNiao/6153) of Filtered Water returns
```
{
  "dcName": "陆行鸟",
  "entries": [],
  "itemID": 6153,
  "lastUploadTime": 1593837555265,
  "stackSizeHistogram": {},
  "stackSizeHistogramNQ": {},
  "stackSizeHistogramHQ": {},
  "regularSaleVelocity": 0,
  "nqSaleVelocity": 0,
  "hqSaleVelocity": 0
}
```
But market history records do exist, as querying [Market Listing API](https://universalis.app/api/LuXingNiao/6153) of Filtered Water returns parts of the market history records in `recentHistory`.
Querying Market History API with other IDs ( such as 5143 or 7153 ) could get normal results.
I'm not sure if this bug is persistent. At least it occurs for now.

**Comment by ha-ku** [comment]
Bug still exists. Any investigation done?

**Comment by karashiiro** [comment]
Not yet, I’ve been busy tackling feature requests on something else I’ve been getting bugged about. I had a quick look at it when this issue was opened, because I was expecting to see a duplicate record, but I didn’t find anything out of the ordinary. I’ll look into this more later.

**Comment by karashiiro** [comment]
The same issue occurs on item [6152](https://universalis.app/api/history/LuXingNiao/6152).

I deleted the database entries for both of those - can you try doing an upload and seeing if they update correctly after that?

**Comment by ha-ku** [comment]
Item 6152 seems working fine for me. I follow your link and it returns something like
`
{"dcName":"陆行鸟","entries":[{"hq":true,"pricePerUnit":555,"quantity":3,"timestamp":1625323374,"worldName":"ShenYiZhiDi"},{"hq":false,"pricePerUnit":665,"quantity":99,"timestamp":1625306916,"worldName":"ShenYiZhiDi"},{"hq":true,"pricePerUnit":1288,"quantity":3,"timestamp":1625304694,"worldName":"ShenYiZhiDi"},{"hq":true,"pricePerUnit":600,"quantity":2,"timestamp":1625304196,"worldName":"ShenYiZhiDi"},{"hq":true,"pricePerUnit":1499,"quantity":15,"timestamp":1625288928,"worldName":"ShenYiZhiDi"},{"hq":false,"pricePerUnit":665,"quantity":99,"timestamp":1625249901,"worldName":"ShenYiZhiDi"},{"hq":false,"pricePerUnit":780,"quantity":48,"timestamp":1625239785,"worldName":"ShenYiZhiDi"},{"hq":false,"pricePerUnit":6500,"quantity":1,"timestamp":1625229194,"worldName":"ShenYiZhiDi"},...
`

For item 6153 (filtered water), I visited the market in game with the universalis act plugin turned on. The web frontend shows that the data was uploaded ( the most recent upload time appeared to be several seconds ago ), but [the api I mentioned above](https://universalis.app/api/history/LuXingNiao/6153)  still returns empty result after that upload.

**Comment by karashiiro** [comment]
I made a modification to the upload code, can you try it one more time now and see if that fixes it?

**Comment by ha-ku** [comment]
I uploaded several times but the[ history api](https://universalis.app/api/history/LuXingNiao/6153) still returns empty results.

**Comment by karashiiro** [comment]
I haven’t been able to figure it out yet, so I’ll begin implementing a bigger database change that I’ve been planning for a while to fix this, removing the DC info and just building this from the world info. Should be done in a couple of days at most.

**Comment by ha-ku** [comment]
Does this change mean that the structure of the return value of the history api will also change?

**Comment by karashiiro** [comment]
It will not have a different structure, but I might have more entries on it depending on how fast it ends up being.

**Comment by ha-ku** [comment]
Okay. Thanks for your hard work!

**Comment by karashiiro** [comment]
Implemented through #447 #448 #449 - should be fixed now, how does it look?

**Comment by ha-ku** [comment]
It works fine now. Thanks!
